django/core/handlers/asgi.py

```diff
@@ -223,6 +227,17 @@ async def process_request(request, send):
                     pass
         body_file.close()
 
+        if response is not None:
+            # If response exists, we must ensure `close` gets called. Do not allow this
+            # to be cancelled
+            await asyncio.shield(sync_to_async(response.close, thread_sensitive=True)())
+        else:
+            # The request processing could have been cancelled before the response
+            # object is created. We still have to fire the request_finished signal to
+            # close resources that might have been opened while the request was
+            # processing
+            signals.request_finished.send(sender=self)
```

Comment:
This should probably use `asend` no? 

Q: Would this be better placed in the `# Ignore client disconnects.` exception handler (where we currently `pass`)? 🤔

Reply:
`asend` - yes - I've made that update and also modified the tests so they are more strict about this